ssl/statem/statem_lib.c

```diff
@@ -113,6 +113,11 @@ int tls_setup_handshake(SSL *s)
         }
         if (SSL_IS_FIRST_HANDSHAKE(s)) {
             s->ctx->stats.sess_accept++;
+        } else if ((s->options & SSL_OP_NO_RENEGOTIATION)) {
+            /* Renegotiation is disabled */
+            SSLerr(SSL_F_TLS_SETUP_HANDSHAKE, SSL_R_NO_RENEGOTIATION);
+            ssl3_send_alert(s, SSL3_AL_WARNING, SSL_AD_NO_RENEGOTIATION);
```

Comment:
Yes, read_bytes should take care of this scenario. I was unable to cover this in my testing.
This is certainly less of an issue as the data structures are opaque.
I will note the case immediately below this should never happen either, as it's also taken care of in the record layer; so the new code handles it the same way (although not fatally). The NO_RENEGOTIATION alert is always a WARNING, so it'd have to be a different alert.